pymongo/topology_description.py

```diff
@@ -324,6 +324,14 @@ def apply_selector(
             description = self.server_descriptions().get(address)
             return [description] if description else []
 
+        # Primary selection fast path.
+        if self.topology_type == TOPOLOGY_TYPE.ReplicaSetWithPrimary and isinstance(
+            selector, Primary
+        ):
+            for sd in self._server_descriptions.values():
+                if sd.server_type == SERVER_TYPE.RSPrimary:
+                    return [sd]
```

Comment:
should this block also have a fast path for returning `[]` when there is no primary?